src/lucky/route_handler.cr

```diff
@@ -4,7 +4,21 @@ class Lucky::RouteHandler
   include HTTP::Handler
 
   def call(context : HTTP::Server::Context)
-    handler = Lucky.router.find_action(context.request)
+    original_path = context.request.path
+
+    # Extract format from URL path and strip it for route matching
+    if url_format = Lucky::MimeType.extract_format_from_path(original_path)
+      context._url_format = url_format
+      # Create a modified request with format-stripped path for route matching
+      path_without_format = original_path.sub(/\.[a-zA-Z0-9]+(?:\?.*)?$/, "")
+      modified_request = context.request.dup
```

Comment:
Is there any considerations of memory here if say you're uploading a 5GB video file in the request? :grimacing:

Reply:
I'm fairly certain the .dup here only performs a shallow copy, so it should be a matter of a few kilobytes and not the file content. For large uploads, the file should already be on disk as a temporary file, and only the IO reference should be copied.

Reply:
Ok cool. I asked in the Crystal discord for clarification, but we can roll with this for now and just keep in mind in case this comes up.